### PR TITLE
BE-64 Add greedy string tiling comparison

### PR DESCRIPTION
**Pull request by Scr0ll**

### CODE REVIEW

File: antiplag/models.py [hunk 1]
```diff
@@ -51,3 +51,7 @@ def __str__(self):
     @property
     def matches(self):
         return len(self.matched_docs)
+
+    @property
+    def intervals(self):
+        return self.matched_docs[0]["intervals"]
```
Comment (MCFreddie777) [inline antiplag/models.py:57]: Property `intervals` of Result only returns intervals of first matched document? I don't get the logic behind it. What about other matched_docs in the list?

Comment (MCFreddie777) [inline antiplag/models.py:57]: Missing empty line.

File: nlp/text_comparison.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+    second_to_first = gst(b, a, length)
+
+    return {"first_to_second": {"intervals": first_to_second[0], "similarity": round(first_to_second[1] / len(a), 3)},
+           "second_to_first": {"intervals": second_to_first[0], "similarity": round(second_to_first[1] / len(b), 3)}}
 
 
 def text_comparison(doc1, doc2):
-    document_vectors = TfidfVectorizer().fit_transform([doc1, doc2])
-    return cosine_similarity(document_vectors)[0][1]
+    return compare_two_texts(doc1, doc2, 50)
```
Comment (MCFreddie777) [inline nlp/text_comparison.py:13]: I think the length `50` - should be stored somewhere in configuration file, so we can easily modify this value if we wanted to

**Review by MCFreddie777 — CHANGES_REQUESTED** [review]

File: antiplag/models.py (hunk 1 above)
Comment (Scr0ll) [reply antiplag/models.py]: fixed

File: nlp/text_comparison.py (hunk 2 above)
Comment (Scr0ll) [reply nlp/text_comparison.py]: added constant to setting.py

File: antiplag/models.py (hunk 1 above)
Comment (Scr0ll) [reply antiplag/models.py]: fixed. returning intervals for each doc separately.

File: antiplag/models.py [hunk 3]
```diff
@@ -37,6 +37,10 @@ class DocumentType(models.TextChoices):
     def __str__(self):
         return f"{self.name if self.name else f'Document {self.id}'} ({self.type})"
 
+    @property
+    def intervals(self):
+        return None
+
```
Comment (OliverKanat) [inline antiplag/models.py:43]: What is the point of this property if it returns None everytime?

File: antiplag/models.py [hunk 4]
```diff
@@ -51,3 +55,6 @@ def __str__(self):
     @property
     def matches(self):
         return len(self.matched_docs)
+
+
+
```
Comment (OliverKanat) [inline antiplag/models.py:60]: Unnecessary empty lines. Only 1 empty line is needed.

File: django_project/settings.py [hunk 5]
```diff
@@ -140,3 +140,6 @@
 
 # Set max files in a request
 MAX_FILES_PER_REQUEST = 50
+
+#Minimal string length considered similarity
+SIMILARITY_LENGTH = 50
```
Comment (OliverKanat) [inline django_project/settings.py:145]: I would suggest adding `MIN_` prefix for this variable.

File: antiplag/serializers.py [hunk 6]
```diff
@@ -56,7 +56,7 @@ class DocumentDetailedSerializer(NonNullModelSerializer, serializers.ModelSerial
 
     class Meta:
         model = Document
-        fields = ("id", "name", "matches", "percentage")
+        fields = ("id", "name", "matches", "percentage", "intervals")
```
Comment (OliverKanat) [inline antiplag/serializers.py:59]: I see the point of intervals variable now, but it still returns `None`

Was it part of task to update the endpoints?

Does it skip the `intervals` variable, when you want to deserialize data?

File: django_project/settings.py (hunk 5 above)
Comment (MCFreddie777) [reply django_project/settings.py:145]: And add space between `#` character and first letter of the commentary

File: antiplag/serializers.py (hunk 6 above)
Comment (Scr0ll) [reply antiplag/serializers.py:59]: it assures that no intervals get returned from api/submissions/{id} but gets returned for matched_docs from api/documents/{id}

File: django_project/settings.py (hunk 5 above)
Comment (Scr0ll) [reply django_project/settings.py:145]: done

**Pull request merged**